SDK Showcase/SDK Showcase/Interactors/SDKInteractor.swift

```diff
@@ -239,122 +249,36 @@ extension SDKInteractorReal {
         }
         
         let mobileAuthRequest = PendingMobileAuthRequestEntity(pendingTransaction: pendingTransaction, delegate: self)
-        mobileAuthQueue.enqueue(mobileAuthRequest)
-        completionHandler([.sound])
+        appState.pendingTransactions.insert(mobileAuthRequest)
         appState.setSystemInfo(string: "Push notification handled.")
-    }
-}
-
-//MARK: - UserProfile formatting
-extension SDKInteractorReal {
-    func formatCategoryName(userId: String, authenticatorName: String) -> String {
-        return userId+"-"+authenticatorName
+        completionHandler([.sound])
     }
     
-    func unformatCategoryName(_ formattedName: String) -> (userId: String, authenticatorName: String) {
-        let parts = formattedName.split(separator: "-")
-        return (String(parts[0]), String(parts[1]))
-    }
-}
-
-//MARK: - ChangePinDelegate
-extension SDKInteractorReal: ChangePinDelegate {
-    func userClient(_ userClient: any OneginiSDKiOS.UserClient, didReceivePinChallenge challenge: any OneginiSDKiOS.PinChallenge) {
-        pinPadInteractor.setPinChallenge(challenge)
-        if let _ = challenge.error {
-            appState.setSystemInfo(string: "Wrong previous PIN, please try again (\(challenge.remainingFailureCount))")
+    func handlePendingTransaction(id: String) {
+        guard let request = appState.pendingTransactions.filter({ $0.pendingTransaction?.transactionId == id }).first else {
```

Comment:
`appState.pendingTransactions.first { $0.pendingTransaction?.transactionId == id }` - isn't enough?

Reply:
the other way around :)